pkg/controller/extendeddaemonset/extendeddaemonset_controller.go

```diff
@@ -352,14 +352,14 @@ func (r *ReconcileExtendedDaemonSet) selectNodes(logger logr.Logger, daemonsetSp
 		if len(daemonsetSpec.Strategy.Canary.NodeAntiAffinityKeys) != 0 {
 			for _, node := range nodeList.Items {
 
-				antiAffinityKeysValueString := antiAffinityKeysValue(&node, daemonsetSpec)
-				if _, found := antiAffinityKeysValues[antiAffinityKeysValueString]; !found {
-					antiAffinityKeysValues[antiAffinityKeysValueString] = 0
+				antiAffinityKeysValue := getAntiAffinityKeysValue(&node, daemonsetSpec)
```

Comment:
nit suggestion: perhaps for clarity, we could rename the `antiAffinityKeysValues` map to something less similar to `antiAffinityKeysValue` ?